src/test/java/org/jvnet/hudson/test/JenkinsRuleTest.java

```diff
@@ -127,6 +142,206 @@ public void testTokenHelperMethods() throws Exception {
         makeRequestAndAssertLogin(wc, "charlotte");
     }
 
+    @Test
+    public void getJSONTests() throws Exception {
+
+        // Testing a simple GET that should answer 200 OK and a json
+        JenkinsRule.JSONWebResponse response = j.getJSON("testing-cli/getMyJsonObject");
+        assertTrue(response.getContentAsString().contains("I am JenkinsRule"));
+        assertEquals(response.getStatusCode(), 200);
+
+        // Testing a simple GET with parameter that should answer 200 OK and a json
+        response = j.getJSON("testing-cli/getWithParameters?paramValue=whatelse");
+        assertTrue(response.getContentAsString().contains("I am JenkinsRule whatelse"));
+        assertEquals(response.getStatusCode(), 200);
+
+        //Testing with a GET that the test expect to raise an server error: we want to be able to assert the status
+        JenkinsRule.WebClient webClientAcceptException = j.createWebClient();
+        webClientAcceptException.setThrowExceptionOnFailingStatusCode(false);
+        response = j.getJSON("testing-cli/getError500", webClientAcceptException);
+        assertEquals(response.getStatusCode(), 500);
+
+        //Testing a GET that requires the user to be authenticated
+        User admin = User.getById("admin", true);
+        MockAuthorizationStrategy auth = new MockAuthorizationStrategy()
+                .grant(Jenkins.ADMINISTER).everywhere().to(admin);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(auth);
+
+        // - simple call without authentication should be forbidden
+        response = j.getJSON("testing-cli/getMyJsonObject", webClientAcceptException);
+        assertEquals(response.getStatusCode(), 403);
+
+        // - same call but authenticated should be fine
+        response = j.getJSON("testing-cli/getMyJsonObject", webClientAcceptException.withBasicApiToken(admin));
```

Comment:
Or this (which we do _not_ expect to throw an exception so why are we using `webClientAcceptException`?) would better be expressed as
```suggestion
        response = j.createWebClient().withBasicApiToken(admin).getJSON("testing-cli/getMyJsonObject");
```